BibBuddy/app/src/main/java/de/bibbuddy/SearchModel.java

```diff
@@ -37,35 +35,52 @@ public SearchModel(Context context) {
    *
    * @return Returns the searchResultList
    */
-  public List<SearchItem> getSearchResultList(String searchInput, SearchSortCriteria sortCriteria) {
+  public List<SearchItem> getSearchResultList(String searchInput, SearchSortCriteria sortCriteria,
+                                              boolean[] filterCriteria) {
     searchResultList.clear();
 
-    // TODO filter
+    if (filterCriteria[0]) { // filter shelf
+      searchShelves(searchInput);
+    }
+
+    if (filterCriteria[1]) { // filter book
+      searchBooks(searchInput);
+    }
+
+    if (filterCriteria[2]) { // filter note
```

Comment:
And here, please remove the unnecessary brackets in order to keep the code clean and simple.